forc-plugins/forc-client/src/ops/deploy/op.rs

```diff
@@ -72,3 +54,31 @@ pub async fn deploy(command: DeployCommand) -> Result<fuel_tx::ContractId> {
         Err(e) => bail!("{e}"),
     }
 }
+
+fn build_opts_from_cmd(cmd: &DeployCommand) -> pkg::BuildOpts {
+    pkg::BuildOpts {
+        pkg: pkg::PkgOpts {
+            path: cmd.path.clone(),
+            offline: cmd.offline_mode,
+            terse: cmd.terse_mode,
+            locked: cmd.locked,
+            output_directory: cmd.output_directory.clone(),
+        },
+        print: pkg::PrintOpts {
+            ast: cmd.print_ast,
+            finalized_asm: cmd.print_finalized_asm,
+            intermediate_asm: cmd.print_intermediate_asm,
+            ir: cmd.print_ir,
+        },
+        minify: pkg::MinifyOpts {
+            json_abi: cmd.minify_json_abi,
+            json_storage_slots: cmd.minify_json_storage_slots,
+        },
+        build_profile: cmd.build_profile.clone(),
+        release: cmd.release,
+        time_phases: cmd.time_phases,
+        binary_outfile: cmd.binary_outfile.clone(),
+        debug_outfile: cmd.debug_outfile.clone(),
+        tests: false,
+    }
+}
```

Comment:
nit: would we be able to share `build_opts_from_cmd` in the root of `forc-client` since it seems to be duplicated between both `deploy` and `run`, or is there a chance they could differ in future? Still going to give the tick regardless!

Reply:
Unfortunately while the run and deploy commands both look the same, they're different types and so need different functions at the moment! That said, if we can share the common build arguments (as this PR starts doing between the build and test commands) then we could construct the BuildOpts from the shared build arguments. I almost did that for this PR, but didn't want to introduce `forc` as a dependency to `forc-client` without giving it more thought. I'm thinking we can possibly split out the CLI argument types into a `forc-cli-args` crate that can be shared, but I might leave that for future work.